### PR TITLE
jiradozer: add local file-backed tracker for --description mode

### PR DESCRIPTION
**Pull request by mzhaom**
## Summary

- Adds a `local` tracker backend (`jiradozer/tracker/local/`) that persists issue state to JSON files on disk, eliminating the need for Linear or any external tracker
- New `--description` and `--description-file` CLI flags create a local issue and run the full plan→build→validate→ship workflow
- Title auto-generated from description using haiku model; all steps auto-approve by default in local mode

## Usage

```bash
# From a flag
jiradozer --description "Create a hello.txt file containing 'hello world'" --work-dir /tmp/test

# From a file
jiradozer --description-file task.md --work-dir /tmp/test

# From stdin
echo "Fix the login bug" | jiradozer --description-file - --work-dir /tmp/test
```

## Test plan

- [x] Unit tests for local tracker (10 tests covering CRUD, persistence, state filtering, comments)
- [x] All existing tests pass (`bazel test //...` — 52/52)
- [x] Lint passes (`scripts/lint.sh`)
- [ ] Manual E2E: `jiradozer --description "..." --work-dir /tmp/test --verbose`

---

> [!NOTE]
> **Medium Risk**
> Introduces a new tracker backend and changes CLI/config validation paths, which could affect workflow execution and mode selection if edge cases aren’t covered. Risk is mitigated by focused unit tests for persistence/state transitions and minimal changes to existing linear flow.
> 
> **Overview**
> Enables running the plan→build→validate→ship workflow from a plain text task description by adding a new `tracker/local` backend that persists issues/comments as JSON files under `<work-dir>/.jiradozer/`.
> 
> Updates the CLI to support `--description`/`--description-file`, enforce mutual exclusivity with `--issue`/`--team`, default to auto-approve in local mode, and create a local issue before starting the workflow; `createTracker()` now supports `tracker.kind: local`.
> 
> Relaxes config loading/validation so `tracker.api_key` env expansion and required checks are skipped for local tracker usage, and updates docs/README plus Bazel targets and tests to cover the new local tracker.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 400d6071899d72b85404417c5a9a9454e5338f02. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: jiradozer/cmd/jiradozer/main.go [hunk 1]
```diff
@@ -114,10 +125,53 @@ func run(ctx context.Context, args runArgs) error {
...
+	// (config file is optional).
+	var cfg *jiradozer.Config
+	if args.description != "" {
+		cfg = jiradozer.DefaultConfigForTest()
+		cfg.Tracker.Kind = "local"
+		// Try loading config file for overrides but don't fail if missing.
+		if loaded, loadErr := jiradozer.LoadConfig(args.configPath); loadErr == nil {
+			cfg = loaded
+		}
+		cfg.Tracker.Kind = "local"
```
[the remark this answers is not in the document]
Comment (mzhaom) [reply jiradozer/cmd/jiradozer/main.go]: Fixed: clearing cfg.Source.Team when --description is set, so a config file with source.team won't conflict with description mode.
